src/shogun/machine/GeneralizedLinearMachine.h

```diff
@@ -0,0 +1,215 @@
+/*
+ * STILL A WORK IN PROGRESS
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Tej Sukhatme
+ */
+
+
+#include <shogun/lib/config.h>
+
+#include <shogun/lib/common.h>
+#include <shogun/machine/Machine.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/lib/SGMatrix.h>
+
+
+namespace shogun
+{
+    enum GLM_DISTRIBUTION
+	{
+        //TODO GAUSSIAN,
+        //TODO BINOMIAL,
+        //TODO GAMMA,
+        //TODO SOFTPLUS,
+        //TODO PROBIT,
+        POISSON
+	};
+
+class DotFeatures;
+class Features;
+class RegressionLabels;
+
+/** @brief Class for estimating regularized generalized linear models (GLM).
+ *   The regularized GLM minimizes the penalized negative log likelihood:
+ *      .. math::
+ *
+ *		\\min_{\\beta_0, \\beta} \\frac{1}{N}
+ *		\\sum_{i = 1}^N \\mathcal{L} (y_i, \\beta_0 + \\beta^T x_i)
+ *		+ \\lambda [ \\frac{1}{2}(1 - \\alpha) \\mathcal{P}_2 +
+ *						\\alpha \\mathcal{P}_1 ]
+ *
+ *	where :math:`\\mathcal{P}_2` and :math:`\\mathcal{P}_1` are the generalized
+ *	L2 (Tikhonov) and generalized L1 (Group Lasso) penalties, given by:
+ *
+ *	.. math::
+ *
+ *		\\mathcal{P}_2 = \\|\\Gamma \\beta \\|_2^2 \\
+ *		\\mathcal{P}_1 = \\sum_g \\|\\beta_{j,g}\\|_2
+ *
+ *	where :math:`\\Gamma` is the Tikhonov matrix: a square factorization
+ *	of the inverse covariance matrix and :math:`\\beta_{j,g}` is the
+ *	:math:`j` th coefficient of group :math:`g`.
+ *
+ *	The generalized L2 penalty defaults to the ridge penalty when
+ *	:math:`\\Gamma` is identity.
+ *
+ *	The generalized L1 penalty defaults to the lasso penalty when each
+ *	:math:`\\beta` belongs to its own group.
+ *
+ * */
+class GeneralizedLinearMachine : public Machine
+{
+ 	public:
+ 		/** default constructor */
+ 		GeneralizedLinearMachine();
+
+ 		/** destructor */
+ 		virtual ~GeneralizedLinearMachine();
+
+ 		/** copy constructor */
+ 		GeneralizedLinearMachine(const std::shared_ptr<GeneralizedLinearMachine>& machine);
+
+		/** get w
+		 *
+		 * @return weight vector
+		 */
+		virtual SGVector<float64_t> get_w() const;
+
+		/** set w
+		 *
+		 * @param src_w new w
+		 */
+		virtual void set_w(const SGVector<float64_t> src_w);
+
+		/** set bias
+		 *
+		 * @param b new bias
+		 */
+		virtual void set_bias(float64_t b);
+
+		/** get bias
+		 *
+		 * @return bias
+		 */
+		virtual float64_t get_bias() const;
+
+ 		/** set features
+ 		 *
+ 		 * @param feat features to set
+ 		 */
+ 		virtual void set_features(std::shared_ptr<DotFeatures> feat);
+
+		/**
+		* A specialization of the train_machine method
+		* @param feats training data
+		*/
+		virtual bool train_machine(std::shared_ptr<const DenseFeatures<float64_t>> feats);
+
+		/** apply linear machine to data
+		 * for regression problem
+		 *
+		 * @param data (test)data to be classified
+		 * @return classified labels
+		 */
+		virtual std::shared_ptr<RegressionLabels> apply_regression(std::shared_ptr<Features> data=NULL);
```

Comment:
use `override` keyword that way you'll know if the function actually is properly overriding a method...